### PR TITLE
Support local network

### PR DESCRIPTION
**Pull request by evilpeach**
## Describe your changes
- Move all constants, API, indexer, and configures to the config file that can be accessed by `chainId`.
- Getting supported chains by retrieving from `NEXT_PUBLIC_SUPPORTED_CHAIN_IDS` as follows
```
NEXT_PUBLIC_SUPPORTED_CHAIN_IDS=osmosis-1,osmo-test-4,osmo-test-5
```
- Update app context to support this
- Remove all `isMainnet`, `isTestnet` concept.

### How to add your network
1. add your network config here `config/index.ts` like this
```ts
"osmosis-1": {
    chain: "osmosis",
    registryChainName: "osmosis",
    prettyName: "Osmosis",
    lcd: "https://lcd.osmosis.zone",
    rpc: "https://rpc.osmosis.zone",
    indexer: "https://osmosis-mainnet-graphql.alleslabs.dev/v1/graphql",
    api: "https://celatone-api.alleslabs.dev",
    features: {
      faucet: {
        enabled: false,
      },
      wasm: {
        enabled: true,
        storeCodeMaxFileSize: 800_000,
        clearAdminGas: 50_000,
      },
      pool: {
        enabled: true,
      },
    },
    gas: {
      gasPrice: {
        tokenPerGas: 0.025,
        denom: "uosmo",
      },
      gasAdjustment: 1.5,
      maxGasLimit: 25_000_000,
    },
    exampleAddresses: {
      user: "osmo14wk9zecqam9jsac7xwtf8e349ckquzzlx9k8c3" as HumanAddr,
      contract:
        "osmo1p0pxllmqjgl2tefy7grypt34jdpdltg3ka98n8unnl322wqps7lqtu576h" as ContractAddr,
      validator:
        "osmovaloper1hh0g5xf23e5zekg45cmerc97hs4n2004dy2t26" as ValidatorAddr,
    },
    explorerLink: {
      validator: "https://www.mintscan.io/osmosis/validators",
      proposal: "https://www.mintscan.io/osmosis/proposals",
    },
  }
```
2. append your chain id in `NEXT_PUBLIC_SUPPORTED_CHAIN_IDS`

### CODE REVIEW

File: src/lib/pages/proposal/constants.tsx [hunk 1]
```diff
@@ -1,115 +1,112 @@
...
+      page: storeCodePage,
+      title: "Do I need to submit Proposal to store code in testnet?",
+      description: (
+        <span>
+          On the {chainName}, you can store code without submitting a proposal
+          by using
+          <Link
+            href="/deploy"
+            onClick={() =>
+              AmpTrackUseRightHelperPanel(storeCodePage, "to-/deploy")
```
Comment (evilpeach) [inline src/lib/pages/proposal/constants.tsx:88]: @poomthiti I'm not sure about this why use `to-/deploy`?

Comment (poomthiti) [reply src/lib/pages/proposal/constants.tsx:88]: maybe trying to follow a kebab case, like to-[pathname]

**Review by poomthiti — APPROVED** [review]
LGTM Overall

File: src/lib/pages/proposal/constants.tsx [hunk 2]
```diff
@@ -1,115 +1,112 @@
...
+    },
+    permissionless: {
+      page: storeCodePage,
+      title: "Do I need to submit Proposal to store code in testnet?",
+      description: (
+        <span>
+          On the {chainName}, you can store code without submitting a proposal
+          by using
+          <Link
+            href="/deploy"
```
Comment (bkioshn) [inline src/lib/pages/proposal/constants.tsx:86]: I think we have to `/{chainId}/deploy`

**Pull request merged**